### ISSUE TITLE
ramp down, failure conditions and interfaces

### ISSUE THREAD

**Issue by 31415us** [cvra/molly-the-motion-planner#4]
so the basic idea is that molly generates a simple ramp down trajectory when no valid path is to be found, which should be simple enough. there are nevertheless some edge cases where a path can be generated but for example the end velocity constraint can not be satisfied (i.e: you choose a point that is very near to the robot but you have a high initial velocity meaning that you can get to the target but not break in time). i would call these second constraint violations 'soft failures' as they do not always lead to possible collisions. i think in case of failure molly should communicate the nature of the failure to the caller

tl;dr : we should discuss the molly's interface

**Comment by SyrianSpock** [comment]
The idea is to call molly through a service call initiated by a 'Coordination' ROS node. 
Basically the idea is to run molly on a ROS node that hosts a service server. We can only request a trajectory from it. So in the trajectory message we should provide a way to inform the service client that there is no viable solution, this can be done via an error flag / validity flag in the trajectory message structure.
The [actual message structure proposed](https://github.com/cvra/roscvra/blob/master/cvra_msgs/msg/BaseTrajectory.msg) doesn't contain such a flag. It's an array of ['Points'](https://github.com/cvra/roscvra/blob/master/cvra_msgs/msg/BaseTrajectoryPoint.msg) which contain a timestamp, a pose vector (x,y,theta) and a twist vector (x', v', theta'). We will probably remove the timestamps from each point and just put the sampling period in the Trajectory message.

**Comment by 31415us** [comment]
ok i'd propose an error code enum instead of a simple boolean flag, so we can distinguish different possible failure states, this can/should be useful for recovery

**Comment by SyrianSpock** [comment]
So something like a status flag with defined states (for example http://docs.ros.org/api/sensor_msgs/html/msg/NavSatStatus.html) ?
For the service call, I was thinking something like this docs.ros.org/api/nav_msgs/html/srv/GetPlan.html : a node initiates a service call and provides two points (start and goal) then molly responds with a trajectory.

**Comment by 31415us** [comment]
yeah some predefined failure states is what i had in mind.
hm additionally to that start and goal i'd need a initial heading / velocity vector since this version of the algo handles that too. the question was more what the format of the response will be.
